Project 15, Ingrid H

ingrid2019-ism opened the pull request:
Link to my deployed project: I had big troubles with the deployment on Netlify. Tried to follow the Netlify deploy log but couldn´t get it to work...

My reflections on how this week's project turned out: I liked the project! Wish I had more time to add some extra features like local storage.

Things I'd like to have clarified or explained in more detail:

HIPPIEKICK commented on code/src/App.js:
```diff
@@ -1,9 +1,11 @@
     <div>
-      Find me in src/app.js!
+      <Home />
     </div>
```
Since you're only returning one component here, you don't need to wrap it in a div.

HIPPIEKICK commented on code/src/components/ShowDate.js:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <>
+      <h1>{today}</h1>
+    </>
```
Same here, you shouldn't need the fragment since you're only returning one element.

HIPPIEKICK commented on code/src/components/ShowDate.js:
```diff
@@ -0,0 +1,14 @@
+
+export const ShowDate = () => {
+
+  const d = new Date()
+  const today = d.toDateString()
```
Nice with a date!

HIPPIEKICK commented on code/src/components/TodoSummery.js:
```diff
@@ -0,0 +1,18 @@
+
+export const TodoSummery = () => {
+
+  const list = useSelector(store => store.todos.list)
```
Since you're only using the items-array, it's better to get only that:
`const items = useSelector(store => store.todos.list.items)`

HIPPIEKICK commented on code/src/components/TodoInput.js:
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useState } from "react"
```
You can import these on the same row, since they have the same source:
`import React, { useState } from 'react'`

HIPPIEKICK commented on code/src/components/TodoInput.js:
```diff
@@ -0,0 +1,37 @@
+      description: inputValue,
+      done: false
+    }))
+    setInputValue('')
```
Good that you're clearing the input! Right now you can submit empty todos, so maybe set the input-field to be `required`, or check if the input is empty before actually dispatching.

HIPPIEKICK commented on code/src/components/TodoInput.js:
```diff
@@ -0,0 +1,37 @@
+        Type='submit'
+        className='todo-input-button'
+        value='+'
+      ></input>
```
Don't forget to use semantic HTML - this is a button so why not call it a button? 😉

HIPPIEKICK commented on code/src/components/TodoInput.js:
```diff
@@ -0,0 +1,37 @@
+        onChange={event => setInputValue(event.target.value)}
+        value={inputValue}
+        className='todo-input-text'
+      ></input>
```
Since the input-element doesn't have any children you can just make it self closing instead of with opening and closing tags:
```
<input
  type='text'
  placeholder='Add Todo'
  onChange={event => setInputValue(event.target.value)}
  value={inputValue}
  className='todo-input-text'
/>
```

HIPPIEKICK commented on code/src/components/RemoveAll.js:
```diff
@@ -0,0 +1,16 @@
+
+
+  return (
+    <button className='remove-button'
+      type='button'
+      onClick={() => dispatch(todos.actions.removeAll())}>
+      Clear
+    </button>
```
Nice with a clear-button!

HIPPIEKICK commented on code/src/components/TodoList.js:
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { TodoItem } from 'components/TodoItem'
+import { TodoInput } from 'components/TodoInput'
+import { TodoSummery } from 'components/TodoSummery'
```
Remove unused code

HIPPIEKICK commented on code/src/components/TodoList.js:
```diff
@@ -0,0 +1,22 @@
+
+export const TodoList = () => {
+
+  const list = useSelector(store => store.todos.list)
```
Same here, you're only using the items-array so get only that instead:
`const items = useSelector(store => store.todos.list.items)`

HIPPIEKICK commented on code/src/components/TodoItem.js:
```diff
@@ -0,0 +1,38 @@
+    )
+  }
+
+  return <div className={`todo-item ${item.done ? 'done' : ''}`}>
+    <input type='checkbox'
+      onChange={handleOnChange}
+      className='todo-item-check'
+      checked={item.done ? 'checked' : ''}></input>
+    <span className='todo-item-description'>{item.description}</span>
+    <button className='todo-item-remove' onClick={onRemoveClicked}>-</button>
+  </div >
```
Nice with conditional styling! Try to be consistent with parentheses after return and maybe clean this up a bit (so the `<div>` gets its own line and the `</div>` don't have a space):
```
return (
  <div className={`todo-item ${item.done ? 'done' : ''}`}>
    <input type='checkbox'
      onChange={handleOnChange}
      className='todo-item-check'
      checked={item.done ? 'checked' : ''}>
    </input>
    <span className='todo-item-description'>{item.description}</span>
    <button className='todo-item-remove' onClick={onRemoveClicked}>-</button>
  </div>
)
```

HIPPIEKICK commented on code/src/index.css:
```diff
@@ -1,13 +1,154 @@
+}
+
+
+.header {
+  display: flex;
+  flex-direction: column;
+  }
+
+ h1 {
+   margin-top: 50px;
+   color: rgb(247, 214, 234);
+ }
```
indentation

HIPPIEKICK commented on code/src/index.css:
```diff
@@ -1,13 +1,154 @@
+    background: #fcfff4;
+    background: linear-gradient(top, #fcfff4 0%, #dfe5d7 40%, #b3bead 100%);
+    border-radius: 50px;
+  }
```
indentation

HIPPIEKICK commented on code/src/index.css:
```diff
@@ -1,13 +1,154 @@
+      margin-right: 50px;
+}
+    
+}
```
indentation

HIPPIEKICK commented on code/src/reducers/todos.js:
```diff
@@ -0,0 +1,34 @@
+          index !== itemIndex
+      )
+    },
+    removeAll: (state, action) => {
```
⭐

ingrid2019-ism commented:
Hi!

I have fixed the indentation and changed the input field to a button tag 
:)

------ Originalmeddelande ------
Från: "Matilda Arvidsson" <notifications@github.com>
Till: "Technigo/project-todos" <project-todos@noreply.github.com>
Kopia: "ingrid2019-ism" <ingrid.hammargren@gmail.com>; "Author" 
<author@noreply.github.com>
Skickat: 2020-05-05 15:19:23
Ämne: Re: [Technigo/project-todos] Project 15, Ingrid H (#57)

>
>@HIPPIEKICK requested changes on this pull request.
>
>Good job Ingrid! It's nice to see that you're adding some extra 
>features like date and clear-button. Please have another look at the 
>code guidelines: 
>https://www.notion.so/technigo/Guidelines-for-how-to-write-good-code-69ffb59eac4e4358a7c5752efb873c4e 
>and make sure you're using semantic HTML as much as possible (I made a 
>comment about an input which is a button). I've also made a few 
>comments about indentation. I will approve this as soon as you fix 
>indentation and change the button to be a button - apart from that - 
>well done!
>
>--------------------------------------------------------------------------------
>In code/src/App.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420069210>:
>
> >      <div>
>
>-      Find me in src/app.js!
>
>+      <Home />
>
>      </div>
>
>
>Since you're only returning one component here, you don't need to wrap 
>it in a div.
>
>--------------------------------------------------------------------------------
>In code/src/components/ShowDate.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420070050>:
>
> > +    <>
>
>+      <h1>{today}</h1>
>
>+    </>
>
>
>Same here, you shouldn't need the fragment since you're only returning 
>one element.
>
>--------------------------------------------------------------------------------
>In code/src/components/ShowDate.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420070203>:
>
> > +  const d = new Date()
>
>+  const today = d.toDateString()
>
>
>Nice with a date!
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoSummery.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420071250>:
>
> > @@ -0,0 +1,18 @@
>
>+import React from 'react'
>
>+import { TodoItem } from 'components/TodoItem'
>
>+import { TodoInput } from 'components/TodoInput'
>
>+import { useSelector } from 'react-redux'
>
>+
>
>+export const TodoSummery = () => {
>
>+
>
>+  const list = useSelector(store => store.todos.list)
>
>
>Since you're only using the items-array, it's better to get only that:
>const items = useSelector(store => store.todos.list.items)
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoInput.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420072679>:
>
> > +import React from 'react'
>
>+import { useState } from "react"
>
>
>You can import these on the same row, since they have the same source:
>import React, { useState } from 'react'
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoInput.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420072992>:
>
> > @@ -0,0 +1,37 @@
>
>+import React from 'react'
>
>+import { useState } from "react"
>
>+import { useDispatch } from 'react-redux'
>
>+import { todos } from 'reducers/todos'
>
>+
>
>+export const TodoInput = () => {
>
>+  const dispatch = useDispatch()
>
>+  const [inputValue, setInputValue] = useState('')
>
>+  const handleOnSubmit = (event) => {
>
>+    event.preventDefault()
>
>+    dispatch(todos.actions.addTodo({
>
>+      description: inputValue,
>
>+      done: false
>
>+    }))
>
>+    setInputValue('')
>
>
>Good that you're clearing the input! Right now you can submit empty 
>todos, so maybe set the input-field to be required, or check if the 
>input is empty before actually dispatching.
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoInput.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420074665>:
>
> > +      <input
>
>+        Type='submit'
>
>+        className='todo-input-button'
>
>+        value='+'
>
>+      ></input>
>
>
>Don't forget to use semantic HTML - this is a button so why not call it 
>a button? 😉
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoInput.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420075534>:
>
> > +      <input
>
>+        Type='text'
>
>+        placeholder='Add Todo'
>
>+        onChange={event => setInputValue(event.target.value)}
>
>+        value={inputValue}
>
>+        className='todo-input-text'
>
>+      ></input>
>
>
>Since the input-element doesn't have any children you can just make it 
>self closing instead of with opening and closing tags:
>
><input
>
>   type='text'
>
>   placeholder='Add Todo'
>
>   onChange={event => setInputValue(event.target.value)}
>
>   value={inputValue}
>
>   className='todo-input-text'
>
>/>
>
>
>--------------------------------------------------------------------------------
>In code/src/components/RemoveAll.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420081572>:
>
> > +    <button className='remove-button'
>
>+      type='button'
>
>+      onClick={() => dispatch(todos.actions.removeAll())}>
>
>+      Clear
>
>+    </button>
>
>
>Nice with a clear-button!
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoList.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420082123>:
>
> > +import { TodoInput } from 'components/TodoInput'
>
>+import { TodoSummery } from 'components/TodoSummery'
>
>
>Remove unused code
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoList.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420082676>:
>
> > @@ -0,0 +1,22 @@
>
>+import React from 'react'
>
>+import { TodoItem } from 'components/TodoItem'
>
>+import { TodoInput } from 'components/TodoInput'
>
>+import { TodoSummery } from 'components/TodoSummery'
>
>+import { useSelector } from 'react-redux'
>
>+
>
>+export const TodoList = () => {
>
>+
>
>+  const list = useSelector(store => store.todos.list)
>
>
>Same here, you're only using the items-array so get only that instead:
>const items = useSelector(store => store.todos.list.items)
>
>--------------------------------------------------------------------------------
>In code/src/components/TodoItem.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420085859>:
>
> > +  return <div className={`todo-item ${item.done ? 'done' : ''}`}>
>
>+    <input type='checkbox'
>
>+      onChange={handleOnChange}
>
>+      className='todo-item-check'
>
>+      checked={item.done ? 'checked' : ''}></input>
>
>+    <span className='todo-item-description'>{item.description}</span>
>
>+    <button className='todo-item-remove' onClick={onRemoveClicked}>-</button>
>
>+  </div >
>
>
>Nice with conditional styling! Try to be consistent with parentheses 
>after return and maybe clean this up a bit (so the <div> gets its own 
>line and the </div> don't have a space):
>
>return (
>
>   <div className={`todo-item ${item.done ? 'done' : ''}`}>
>
>     <input type='checkbox'
>
>       onChange={handleOnChange}
>
>       className='todo-item-check'
>
>       checked={item.done ? 'checked' : ''}>
>
>     </input>
>
>     <span className='todo-item-description'>{item.description}</span>
>
>     <button className='todo-item-remove' onClick={onRemoveClicked}>-</button>
>
>   </div>
>
>)
>
>
>--------------------------------------------------------------------------------
>In code/src/index.css 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420087414>:
>
> > +.header {
>
>+  display: flex;
>
>+  flex-direction: column;
>
>+  }
>
>+
>
>+ h1 {
>
>+   margin-top: 50px;
>
>+   color: rgb(247, 214, 234);
>
>+ }
>
>
>indentation
>
>--------------------------------------------------------------------------------
>In code/src/index.css 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420087789>:
>
> > +.todo-item-check {
>
>+  width: 15px;
>
>+    height: 15px;
>
>+    position: relative;
>
>+    margin: 20px auto;
>
>+    margin-right: 5px;
>
>+    background: #fcfff4;
>
>+    background: linear-gradient(top, #fcfff4 0%, #dfe5d7 40%, #b3bead 100%);
>
>+    border-radius: 50px;
>
>+  }
>
>
>indentation
>
>--------------------------------------------------------------------------------
>In code/src/index.css 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420088281>:
>
> > +@media screen and (min-width: 600px) {
>
>+    .todo-item {
>
>+      margin-left: 200px;
>
>+    }
>
>+
>
>+    .input-remove {
>
>+      justify-content: center;
>
>+    }
>
>+
>
>+    .todo-input-button {
>
>+      margin-right: 50px;
>
>+}
>
>+
>
>+}
>
>
>indentation
>
>--------------------------------------------------------------------------------
>In code/src/reducers/todos.js 
><https://github.com/Technigo/project-todos/pull/57#discussion_r420092680>:
>
> > +    addTodo: (state, action) => {
>
>+      const itemInfo = action.payload
>
>+      state.list.items.push(itemInfo)
>
>+    },
>
>+    setDone: (state, action) => {
>
>+      const { itemIndex, done } = action.payload
>
>+      state.list.items[itemIndex].done = done
>
>+    },
>
>+    removeTodo: (state, action) => {
>
>+      const { itemIndex } = action.payload
>
>+      state.list.items = state.list.items.filter(
>
>+        (item, index) =>
>
>+          index !== itemIndex
>
>+      )
>
>+    },
>
>+    removeAll: (state, action) => {
>
>
>⭐
>
>—
>You are receiving this because you authored the thread.
>Reply to this email directly, view it on GitHub 
><https://github.com/Technigo/project-todos/pull/57#pullrequestreview-405737315>, 
>or unsubscribe 
><https://github.com/notifications/unsubscribe-auth/ANTMD66XHBA6BOQKLAOANUDRQAG5XANCNFSM4MP4S3WA>.
>

HIPPIEKICK commented:
Awesome!
